decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/Bcl2fastqDecider.java

```diff
@@ -706,4 +793,91 @@ private Path writeWorkflowRunIniPropertiesToFile(WorkflowRun wr) {
             throw new RuntimeException(ex);
         }
     }
+
+    private List<ProvenanceWithProvider<SampleProvenance>> applyBasesMask(List<ProvenanceWithProvider<SampleProvenance>> samples, BasesMask basesMask) throws DataMismatchException {
+        List<ProvenanceWithProvider<SampleProvenance>> samplesWithCustomBarcodes = new ArrayList<>();
+        for (ProvenanceWithProvider<SampleProvenance> pwp : samples) {
+            SampleProvenance sp = pwp.getProvenance();
+            Barcode newBarcode = BarcodeAndBasesMask.applyBasesMask(Barcode.fromString(sp.getIusTag()), basesMask);
+            SampleProvenanceWithCustomBarcode sampleProvenanceWithCustomBarcode = new SampleProvenanceWithCustomBarcode(sp, newBarcode.toString());
+            ProvenanceWithProvider<SampleProvenance> newPwp = new ProvenanceWithProvider(pwp.getProvider(), sampleProvenanceWithCustomBarcode);
+            samplesWithCustomBarcodes.add(newPwp);
+        }
+        return samplesWithCustomBarcodes;
+    }
+
+    private void addInvalidLane(String message, String... args) {
+        String error = MessageFormat.format(message, (Object[]) args);
+        laneErrors.add(error);
+    }
+
+    private List<WorkflowRunV2> generateWorkflowRunsForLane(
+            ProvenanceWithProvider<LaneProvenance> lp,
+            Map<String, List<ProvenanceWithProvider<SampleProvenance>>> groupedSamples,
+            Bcl2FastqHandler handler,
+            boolean doDemultiplexing) throws InvalidLaneException {
+
+        List<String> laneErrors = new ArrayList<>();
+        List<WorkflowRunV2> workflowRuns = new ArrayList<>();
+        for (Entry<String, List<ProvenanceWithProvider<SampleProvenance>>> e : groupedSamples.entrySet()) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)